### ISSUE TITLE
`flatten` as convinience method for stack all?

### ISSUE THREAD

**Issue by mathause** [pydata/xarray#1029]
There is a bunch of operations that need to be conducted on a 1D array. It could be handy to have a convinience method that mimicks numpys `flatten`, i.e. works along the lines:

```
def flatten(dataarray, name='stacked'):
    return dataarray.stack(**{name: dataarray.dims})
```

What do you think?

**Comment by shoyer** [comment]
Yes, seems like a great idea to me!

**Comment by feldman4** [comment]
This is a good idea, any reason not to add linear indexing with `array.flat[:]` as well?

**Comment by shoyer** [comment]
We might call this `stack_all()` instead -- that's closer to `.stack()` with alphabetical sorting.

**Comment by dcherian** [comment]
Solved by `.stack=[...]`?

**Comment by mathause** [comment]
Yes indeed, good catch.

Closed by #3826
